src/xenia/kernel/xbdm/xbdm_misc.cc

```diff
@@ -257,6 +286,180 @@ dword_result_t DmGetMemory_entry(lpvoid_t src_ptr, dword_t buf_size,
 }
 DECLARE_XBDM_EXPORT1(DmGetMemory, kDebug, kImplemented);
 
+dword_result_t DmGetThreadInfoEx_entry(
+    dword_t thread_id, pointer_t<DM_THREAD_INFO_EX> thread_info) {
+  if (!thread_info) {
+    return X_E_INVALIDARG;
+  }
+
+  thread_info.Zero();
+
+  auto thread = kernel_state()->GetThreadByID(thread_id);
+
+  if (!thread) {
+    return XBDM_UNSUCCESSFUL;
+  }
+
+  uint32_t thread_name_size = static_cast<uint32_t>(
+      xe::string_util::size_in_bytes(thread->name(), true));
+  uint32_t thread_name_address =
+      kernel_state()->memory()->SystemHeapAlloc(thread_name_size);
+
+  char* thread_name_ptr =
+      kernel_state()->memory()->TranslateVirtual<char*>(thread_name_address);
+
+  xe::string_util::copy_truncating(thread_name_ptr, thread->name().data(),
+                                   thread_name_size);
+
+  uint32_t tls_base_address =
+      kernel_state()->memory()->SystemHeapAlloc(sizeof(uint32_t));
```

Comment:
Remember that SystemHeapAlloc allocates page size, so you're basically using 4kb for 16 bytes